tracing/span_test.go

```diff
@@ -0,0 +1,167 @@
+package tracing
+
+import (
+	"context"
+	"testing"
+
+	"github.com/opentracing/opentracing-go"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+)
+
+func TestKeyValueToOTelAttribute(t *testing.T) {
+	type kv struct {
+		k string
+		v any
+	}
+	tests := []struct {
+		name     string
+		kv       kv
+		expected attribute.KeyValue
+	}{
+		{
+			name:     "string",
+			kv:       kv{"key", "value"},
+			expected: attribute.String("key", "value"),
+		},
+		{
+			name:     "int",
+			kv:       kv{"key", 42},
+			expected: attribute.Int("key", 42),
+		},
+		{
+			name:     "int64",
+			kv:       kv{"key", int64(42)},
+			expected: attribute.Int64("key", 42),
+		},
+		{
+			name:     "float64",
+			kv:       kv{"key", 42.0},
+			expected: attribute.Float64("key", 42.0),
+		},
+		{
+			name:     "bool",
+			kv:       kv{"key", true},
+			expected: attribute.Bool("key", true),
+		},
+		{
+			name:     "string slice",
+			kv:       kv{"key", []string{"value1", "value2"}},
+			expected: attribute.StringSlice("key", []string{"value1", "value2"}),
+		},
+		{
+			name:     "int slice",
+			kv:       kv{"key", []int{1, 2, 3}},
+			expected: attribute.IntSlice("key", []int{1, 2, 3}),
+		},
+		{
+			name:     "int64 slice",
+			kv:       kv{"key", []int64{1, 2, 3}},
+			expected: attribute.Int64Slice("key", []int64{1, 2, 3}),
+		},
+		{
+			name:     "stringer",
+			kv:       kv{"key", testStringer{}},
+			expected: attribute.Stringer("key", testStringer{}),
+		},
```

Comment:
nit: does not test `[]byte` and fallback.

Reply:
Done in https://github.com/grafana/dskit/pull/681/commits/0b10eb1182aaa1ec3b1edb3bcfcdeb24018d3d41